src/lib/server/migrations.ts

```diff
@@ -509,6 +509,17 @@ const migrations = [
 				{ session }
 			);
 		}
+	},
+	{
+		_id: new ObjectId('68230a070000000000000000'),
+		name: 'Ensure payment processor is set on lightning payments',
+		run: async (session: ClientSession) => {
+			await collections.orders.updateMany(
+				{ 'payments.method': 'lightning', 'payments.processor': '' },
```

Comment:
```suggestion
				{ 'payments': { $elemMatch: {method: 'lightning', processor: { $exists: false } } } },
```

When we want to add multiple conditions that a specific element in an array needs to match, we need to use `$elemMatch`: https://www.mongodb.com/docs/manual/reference/operator/query/elemMatch/

Also I think the absent processor would be `{$exists: false}`

Reply:
Applied (and tested) suggestion.